### PR TITLE
Add note on fix-up commits

### PR DESCRIPTION
**Pull request by codeinthehole**

### CODE REVIEW

File: git.md [hunk 1]
```diff
@@ -75,3 +75,41 @@ when you want to pull changes from `master` into your branch.
...
+However, this isn't so great for the reviewer who needs to re-review every
+commit once code-review is requested again.
+
+It's better to use temporary "fix-up" commits that the reviewer can easily
+evaluate to see if they address the comments. Once approved, these can easily be
+rebased into the PR branch before merge, achieving the same commit granularity
+requirements but making better use of the code reviewer's time.
+
+These have a special commit subject synxtax
+that git recognises and knows how to rebase them
```
Comment (scalen) [inline git.md]: "that git recognises and uses to orchestrate rebases performed with `--autosquash`."

File: git.md [hunk 2]
```diff
@@ -75,3 +75,41 @@ when you want to pull changes from `master` into your branch.
...
+These have a special commit subject synxtax
+that git recognises and knows how to rebase them 
+
+Rather than rebasing changes into the original PR commits, use "fix-up"
+commits. These are special commits that git knows how to rebase into the branch
+before merge.
+
+tha
+
+- Are easy for the reviewer to
```
Comment (scalen) [inline git.md]: ????????

Comment (codeinthehole) [reply git.md]: I did say WIP

Comment (scalen) [reply git.md]: True :-P I can be a bit over-eager sometimes, I'll try to reign it in!

File: git.md [hunk 3]
```diff
@@ -75,3 +83,36 @@ when you want to pull changes from `master` into your branch.
 Further reading:
 
 - [Advanced pull-request crafting](https://codeinthehole.com/tips/advanced-pull-request-crafting/)
+
+
+<a name="fixup-commits">Use "fix-up" commits to address code review comments</a>
```
Comment (seddonym) [inline git.md]: This should be a subheading.

Comment (codeinthehole) [reply git.md]: Good spot - thanks

**Pull request merged**